static/js/search.js

```diff
@@ -2,9 +2,33 @@ var search = (function () {
 
 var exports = {};
 
+function ValidURL(str)
+{
+    var patt = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!-\/]))?/;
+    if (!patt.test(str)) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+function addDoubleQuotes(str)
+{
+    if (str[0] !== "\"") {
```

Comment:
I would spell this `'"'` rather (with single-quotes quoting the double-quote character) than using the escape character